addons/website_livechat/tests/test_chatbot_ui.py

```diff
@@ -221,3 +221,50 @@ def test_chatbot_fw_operator_matching_lang(self):
         channel = self.livechat_channel.channel_ids[0]
         self.assertIn(channel.channel_member_ids.partner_id.user_ids, en_op)
         self.assertNotIn(channel.channel_member_ids.partner_id.user_ids, fr_op)
+
+
+    def test_question_selection_overlapping_answers(self):
+        chatbot_script = self.env["chatbot.script"].create({"title": "Question selection bot"})
+        question_1 = self.env["chatbot.script.step"].create(
+            [
+                {
+                    "chatbot_script_id": chatbot_script.id,
+                    "message": "Choose an option",
+                    "step_type": "question_selection",
+                },
+            ]
+        )
+        x_answer = self.env["chatbot.script.answer"].create({
+            "name": "X",
+            "script_step_id": question_1.id,
+        })
+        not_x_answer = self.env["chatbot.script.answer"].create({
+            "name": "not X",
+            "script_step_id": question_1.id,
+        })
+        self.env["chatbot.script.step"].create(
+            [
+                {
+                    "chatbot_script_id": chatbot_script.id,
+                    "step_type": "text",
+                    "triggering_answer_ids": [x_answer.id],
+                    "message": "You selected X",
+                },
+                {
+                    "chatbot_script_id": chatbot_script.id,
+                    "step_type": "text",
+                    "triggering_answer_ids": [not_x_answer.id],
+                    "message": "You selected not X",
+                },
+            ]
+        )
+        self.livechat_channel.rule_ids = self.env["im_livechat.channel.rule"].create(
+            [
+                {
+                    "channel_id": self.livechat_channel.id,
+                    "chatbot_script_id": chatbot_script.id,
+                    "regex_url": "/",
+                },
+            ]
+        )
+        self.start_tour("/", "website_livechat.question_selection_overlapping_answers")
```

Comment:
Ok test pass now with changes, when it didn't.

Note that this is not super intuitive to make 2 answers ordered as `X` than `no X`. At first I thought the bug was that it always picked the 1st option matching, but apparently no the bug was the last answer matching.

Without knowing why the test is written in this order, personally I think the test looks a bit stupid. What about making an aswer in-between 2 answers that are longer and contain this answer? This would test both scenario and the test would feel more relevant compared by how this is crafted now.